js/utils/oppfolgingsplanUtils.js

```diff
@@ -155,3 +155,97 @@ export const manglerSamtykke = (oppfolgingsplan) => {
   }
   return oppfolgingsplan.arbeidsgiver.naermesteLeder.samtykke === null;
 };
+
+export const erSykmeldingGyldigForOppfolgingMedGrensedato = (perioder, dato) => {
+  return (
+    perioder.filter((periode) => {
+      const tomGrenseDato = new Date(dato);
+      tomGrenseDato.setHours(0, 0, 0, 0);
+      tomGrenseDato.setMonth(tomGrenseDato.getMonth() - MND_SIDEN_SYKMELDING_GRENSE_FOR_OPPFOLGING);
+      return new Date(periode.tom) >= new Date(tomGrenseDato);
+    }).length > 0
+  );
+};
+
+export const finnSykmeldingerPaVirksomhet = (sykmeldinger, orgnummer) => {
+  return sykmeldinger.filter((sm) => {
+    return sm.orgnummer === orgnummer;
+  });
+};
+
+function compareFomDate(a, b) {
+  if (a.fom < b.fom) {
+    return -1;
+  }
+  if (a.fom > b.fom) {
+    return 1;
+  }
+  return 0;
+}
+
+export const erOppfolgingsplanGyldigForOppfolgingMedGrensedato = (fomLastSykefravar, tomOppfolgingsplan) => {
+  const tomGrenseDato = new Date(
+    fomLastSykefravar.getFullYear(),
+    fomLastSykefravar.getMonth(),
+    fomLastSykefravar.getDate()
+  );
+
+  tomGrenseDato.setHours(0, 0, 0, 0);
+  tomGrenseDato.setMonth(fomLastSykefravar.getMonth() - MND_SIDEN_SYKMELDING_GRENSE_FOR_OPPFOLGING);
+  return new Date(tomOppfolgingsplan) >= tomGrenseDato;
+};
+
+export const finnGyldigePlanerPaVirksomhet = (planer, orgnummer, dineSykmeldteMedSykmeldinger) => {
+  const oppfolgingsplanerPaVirksomhet = finnOppfolgingsplanerPaVirksomhet(planer, orgnummer);
+  const lastSykefravar = getLastSykefravar(dineSykmeldteMedSykmeldinger, orgnummer);
+
+  return oppfolgingsplanerPaVirksomhet.filter((oppfolgingsplan) => {
```

Comment:
Hm, burde du bruke .map her i stedet for filter?

Reply:
Hvorfor? Ingenting blir endret, det returneres liste over den samme samme oppfølgingsplan type. De planene som ikke stiller dato-krav filtreres bort.